python/dgl/frame.py

```diff
@@ -163,8 +184,18 @@ def data(self):
     @data.setter
     def data(self, val):
         """Update the column data."""
-        self.index = None
-        self.storage = val
+        if isinstance(val, tuple):
+            # This case is used to handle DistTensor returned by DistGraph.
+            # index is used to identify rows in DistTensor
+            # that are present in the current graph/block
+            data, index = val
+            assert isinstance(data, distributed.dist_tensor.DistTensor)\
+                , "Input data must be DistTensor when index is specified."
+            self.index = index
+            self.storage = data
```

Comment:
This part should also be gone. The `val` should just be a `DistTensor` instead of a pair.

Reply:
Removed this part and rolled back to the original implementation.